packages/file_selector/file_selector_web/lib/src/dom_helper.dart

```diff
@@ -0,0 +1,51 @@
+import 'dart:async';
+import 'dart:html';
+import 'package:meta/meta.dart';
+import 'package:flutter/services.dart';
+
+/// Class to manipulate the DOM with the intention of reading files from it.
+class DomHelper {
+  final _container = Element.tag('file-selector');
+
+  /// Default constructor, initializes the container DOM element.
+  DomHelper() {
+    final body = querySelector('body');
+    body.children.add(_container);
+  }
+
+  /// Sets the <input /> attributes and waits for a file to be selected.
+  Future<List<File>> getFiles({
+    String accept = '',
+    bool multiple = false,
+    @visibleForTesting FileUploadInputElement input,
+  }) {
+    final Completer<List<File>> _completer = Completer();
+    input = input ?? FileUploadInputElement();
+
+    _container.children.add(
+      input
+        ..accept = accept
+        ..multiple = multiple,
+    );
```

Comment:
can you `..clear()` the `children` before calling `add` with the next input? That way you don't need to remember to call `input.remove` in all the event handlers later.

Reply:
Discussed offline. TLDR both ways are equivalent at this moment. But the current one might be more extensible if, let's say, in the future browser allows more than one dialog to be shown as it is only touching the relevant-for-the-call input (instead of removing the other inputs that could potentially being used).